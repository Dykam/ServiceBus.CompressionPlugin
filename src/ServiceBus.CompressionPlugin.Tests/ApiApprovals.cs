namespace ServiceBus.CompressionPlugin.Tests
{
    using PublicApiGenerator;
    using ServiceBus.CompressionPlugin;
    using Xunit;

    public class ApiApprovals
    {
        [Fact]
        public void CompressionPlugin()
        {
            var publicApi = ApiGenerator.GeneratePublicApi(typeof(CompressionPlugin).Assembly,
                whitelistedNamespacePrefixes: new[] { "Microsoft.Azure.ServiceBus." },
                excludeAttributes: new[] { "System.Runtime.Versioning.TargetFrameworkAttribute" });

            Approver.Verify(publicApi);
        }
    }
}