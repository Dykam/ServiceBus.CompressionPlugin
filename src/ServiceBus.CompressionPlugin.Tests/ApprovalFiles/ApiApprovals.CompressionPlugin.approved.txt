[assembly: System.Runtime.CompilerServices.InternalsVisibleToAttribute("ServiceBus.CompressionPlugin.Tests")]
namespace Microsoft.Azure.ServiceBus
{
    public class CompressionConfiguration
    {
        public CompressionConfiguration(string compressionMethodName, System.Func<byte[], byte[]> compressor, System.Func<byte[], byte[]> decompressor, int minimumSize) { }
    }
    public class static CompressionPluginExtensions
    {
        public static Microsoft.Azure.ServiceBus.Core.ServiceBusPlugin RegisterCompressionPlugin(this Microsoft.Azure.ServiceBus.ClientEntity client) { }
        public static Microsoft.Azure.ServiceBus.Core.ServiceBusPlugin RegisterCompressionPlugin(this Microsoft.Azure.ServiceBus.ClientEntity client, Microsoft.Azure.ServiceBus.CompressionConfiguration compressionConfiguration) { }
    }
}